src/server/scene/surface_stack.cpp

```diff
@@ -609,19 +610,29 @@ void ms::SurfaceStack::lock()
     bool temp = false;
     if (is_locked.compare_exchange_weak(temp, true))
     {
+        mir::log_info("Session is now locked and listeners are being notified");
         emit_scene_changed();
         multiplexer.on_lock();
     }
+    else
+    {
+        mir::log_info("SurfaceStack received lock() but could not compare_exchange_weak");
```

Comment:
...this `log_info` is bad - although that's partially because this should be `compare_exchange_strong`. This *should* be something like:
```suggestion
        mir::log_debug("Session received duplicate lock request");
```